python-frontend/src/main/java/org/sonar/python/metrics/FileLinesVisitor.java

```diff
@@ -121,18 +127,24 @@ private void visitToken(Token token) {
       return;
     }
 
+    linesOfCode.addAll(countTokenLines(token));
+
+    for (Trivia trivia : token.trivia()) {
+      visitComment(trivia, token);
+    }
+  }
+
+  public static Set<Integer> countTokenLines(Token token) {
+    Set<Integer> lines = new HashSet<>();
```

Comment:
Most tokens wouldn't need a `HashSet`. I wonder if this change has a visible impact on performance (e.g. our ruling). Probably not.